tokio/src/signal/windows/sys.rs

```diff
@@ -177,9 +200,9 @@ mod tests {
             // Windows doesn't have a good programmatic way of sending events
             // like sending signals on Unix, so we'll stub out the actual OS
             // integration and test that our handling works.
-            unsafe {
+            std::thread::spawn(|| unsafe {
```

Comment:
Can you add a comment explaining why we do this? (In all three tests.)

Or perhaps you should even have a shared function for calling the handler that checks `event_requires_infinite_sleep_in_handler` and does thread spawn or not depending on the value.

Reply:
Like this?

Reply:
Yes, thanks.